bootstrap/kubeadm/types/v1beta1/utils.go

```diff
@@ -17,12 +17,72 @@ limitations under the License.
 package v1beta1
 
 import (
+	"strings"
+
 	"github.com/pkg/errors"
 	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
+	versionutil "k8s.io/apimachinery/pkg/util/version"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/v1beta2"
 	"sigs.k8s.io/controller-runtime/pkg/scheme"
 )
 
+func KubeVersionToKubeadmAPIGroupVersion(version string) (schema.GroupVersion, error) {
+	if version == "" {
+		return schema.GroupVersion{}, errors.New("version cannot be empty")
+	}
+	semVersion, err := versionutil.ParseSemantic(version)
+	if err != nil {
+		return schema.GroupVersion{}, errors.Wrap(err, "error parsing the Kubernetes version")
+	}
+	switch {
+	case semVersion.LessThan(versionutil.MustParseSemantic("v1.13.0")):
+		return schema.GroupVersion{}, errors.New("the bootstrap provider for kubeadm doesn't support Kubernetes version lower than v1.13.0")
+	case semVersion.LessThan(versionutil.MustParseSemantic("v1.15.0")):
```

Comment:
off topic:
- where can one find the minimum supported version of CAPBK in docs or YAML?
- does CABK error out if the user version is older than the minimum?

Reply:
see https://cluster-api.sigs.k8s.io/reference/versions.html
With this change CABPK will error out for Kubernetes < v1.13, while before things were failing in the cloud init script (which is much more complex to debug).

Reply:
Let's move the checks above to a validation webhook?

Reply:
I will open an issue to discuss this improvement.
The problem here is that this is a CABPK limitation, but the version could be defined on another resource (e.g. Machine deployments)

For the sake of this PR this is a little bit out of topic and this will also deviate from the original PR we are back porting.
Nevertheless we are already improving the status quo, so I hop this should not be a blocker

Reply:
rif https://github.com/kubernetes-sigs/cluster-api/issues/4321